support/docker/production/docker-entrypoint.sh

```diff
@@ -11,6 +11,18 @@ cp /app/config/default.yaml /config
 cp /app/support/docker/production/config/custom-environment-variables.yaml /config
 chown -R peertube:peertube /config
 
+# Parsing trust_proxy if available
+if [ ! -z "${PEERTUBE_TRUST_PROXY}" ]; then
+    # We need to replace space by comma
+    TRUST_PROXY="[${PEERTUBE_TRUST_PROXY// /,}]"
+    # Push the trust_proxy conf to production.yaml and remove old config
+    awk -v var="${TRUST_PROXY}" 's{if(/\s*-\s*.*/) next; else s=0}  /trust_proxy:/{print "trust_proxy: " var; s=1; next} 1' /config/production.yaml > /config/temp-production.yaml
```

Comment:
this seems a bit complex to maintain. I'm also unsure about using a temporary file…

Reply:
Yeah, it's indeed complicated, I have to twist my head to found a proper way to do that.
Th main problem is truxt_proxy is the only option in config using an array, and not a simple strin or true/false value.

It means my previous PR is only useful with one trust_proxy. I have tested a large variety of option, passing to node env variable with [ ], spaces, comma, json formated, but no one are corrects, and worse, node crash when parsing them, causing docker to restart indefinitely...

This is the only solution I found, parsing myself the env variable to create the array, and write it directly inside the configuration file. 

As we use bash, awk seems the most adapted tool, and as we have no modern awk in the docker image, we cannot directly modify the file in place, and need to store the result somewhere before writing again the production.yaml...

More simple bash commands like cat/EOF or cut/trim/echo are worse for modifying an entire file and dealing with all cases :'(

Others options would be to modify the main code of Peertube, like allowing configuring trust_proxy via the web interface, or parsing env variable at start with node, but only for docker needs it seemed even more complicated for me ^^"

Using the entrypoint for parsing env variables are still a good practice for docker, and I tried to be less complex and less space consuming possible. But I could understand it's not trivial.

I am fairly open to any advice about this problem, with gratitude! :hugs:

Reply:
@LecygneNoir I don't have a better option.

Maybe explicit with comments that the said section is only here to support the PEERTUBE_TRUST_PROXY option. This way, if anyone ever bumps into maintainability issues, they can safely remove the incriminated code.

Reply:
Good idea, I'll add more details about this choice :+1: